desktop/cmp/rest/RestGridModel.js

```diff
@@ -174,60 +172,52 @@ export class RestGridModel {
         this.formModel.openClone(clone);
     }
 
-    @action
-    deleteSelection() {
-        const record = this.selectedRecord;
-        if (record) this.deleteRecord(record);
-    }
-
-    @action
-    deleteRecord(record) {
+    async deleteRecord(record) {
         throwIf(this.readonly, 'Record not deleted: this grid is read-only');
         this.store.deleteRecordAsync(record)
             .then(() => this.formModel.close())
             .catchDefault();
     }
 
-    @action
-    editSelection() {
-        const record = this.selectedRecord;
-        if (record) this.editRecord(record);
+    async bulkDeleteRecordsAsync(records) {
+        throwIf(this.readonly, 'Records not deleted: this grid is read-only');
+        const resp = await this.store.bulkDeleteRecordsAsync(records).catchDefault(),
+            intent = resp.fail > 0 ? 'warning' : 'success',
+            message = `Deleted ${resp.success} ${pluralize(this.unit)} with ${resp.fail} failures`;
+
+        XH.toast({intent, message});
     }
 
-    @action
     editRecord(record) {
         this.formModel.openEdit(record);
     }
 
-    @action
     viewRecord(record) {
         this.formModel.openView(record);
     }
 
-    confirmDeleteSelection() {
-        const record = this.selectedRecord;
-        if (record) this.confirmDeleteRecord(record);
-    }
+    confirmDeleteRecords() {
+        const records = this.selection,
+            warning = this.actionWarning.del,
+            delFn = () => records.length > 1 ? this.bulkDeleteRecordsAsync(records) : this.deleteRecord(records[0]);
 
-    confirmDeleteRecord(record) {
-        const warning = this.actionWarning.del;
-        if (warning) {
+        if (!warning) {
+            delFn();
+        } else {
+            const message = isFunction(warning) ? warning(records) : warning;
             XH.confirm({
-                message: warning,
+                message,
                 title: 'Warning',
                 icon: Icon.warning({size: 'lg'}),
-                onConfirm: () => this.deleteRecord(record)
+                onConfirm: () => delFn()
```

Comment:
don't need the extra fat arrow function on 212, can just set to onConfirm: delFn